HerPortal.BusinessLogic/Services/CsvFileService/CsvFileService.cs

```diff
@@ -83,7 +83,7 @@ public async Task<IEnumerable<CsvFileData>> GetFileDataForUserAsync(string userE
     {
```

Comment:
And while we're here I think we can afford to spend some time refactoring to make this method a bit easier to follow, and possibly make the logic easier to test. I find it takes some effort to reason with at the moment: overlong and nested and mutating `files` and referring back to it.

I'm imagining something like
```
var custodianCodes = ...;
var consortiumCodes = ...;

var localAuthoritiesFileData = BuildCsvFileDataForLocalAuthorities(custodianCodes); // the big foreach goes in this method
var consortiaFileData = TransformFileDataForConsortia(files, consortiumCodes);
var combinedFileData = localAuthoritiesFileData.Concat(consortiaFileData);

return combinedFileData
    .OrderBy...
```

Does that make sense?